zenodo/modules/spam/views.py

```diff
@@ -172,6 +182,14 @@ def _expand_users_info(results, include_pending=False):
         for user in user_data
     ])
 
+    email_domains = db.session.query(
+        sa.func.split_part(sa.func.lower(User.email),'@',2),
+        sa.func.count(User.id),
+        sa.func.count(User.id).filter(User.active == True))\
+        .group_by(sa.func.split_part(sa.func.lower(User.email),'@',2)).all()
+    email_domain_count = {domain: {'active': active, 'total': total} for
+                          domain, active, total in email_domains}
```

Comment:
Not a big fan on the formatting here, but I know I'm risking an aggressive AI takeover by saying that :)

I would also breakdown some parts of the query for readability:

```suggestion
    email_domain_expr = sa.func.split_part(sa.func.lower(User.email),'@',2)
    email_domains = db.session.query(
        email_domain_expr,
        sa.func.count(User.id),
        sa.func.count(User.id).filter(User.active == True)
    ).group_by(email_domain_expr).all()
    
    email_domain_count = {
        domain: {'active': active, 'total': total} 
        for domain, active, total in email_domains
    }
```